File: testdata/fixtures/go/CWE-532.go
// CWE-532: Insertion of Sensitive Information into Log File
//
// Logging a password or secret exposes it in log files which are often
// world-readable or shipped to third parties.

package fixture

import (
	"log"
	"os"
)

func leak() {
	password := os.Getenv("PASSWORD")
	secret := os.Getenv("SECRET")
	// ruleid: go-sensitive-log
	log.Println(password)
	// ruleid: go-sensitive-log
	log.Printf("password=%s", password)
	// ruleid: go-sensitive-log
	log.Println(secret)
}

func safe() {
	// ok: go-sensitive-log
	log.Println("user logged in")
	// ok: go-sensitive-log
	log.Printf("request processed")
	// ok: go-sensitive-log
	log.Println("redacted")
}

File: testdata/fixtures/scala/CWE-532.scala
// CWE-532: Insertion of Sensitive Information into Log File
//
// Logging a password or secret exposes it in log files which are often
// world-readable or shipped to third parties.

import org.slf4j.LoggerFactory

object Cwe532Log {
  private val logger = LoggerFactory.getLogger(getClass)

  def leak(password: String, secret: String, apiKey: String): Unit = {
    // ruleid: scala-sensitive-log
    logger.info(password)
    // ruleid: scala-sensitive-log
    logger.debug(secret)
    // ruleid: scala-sensitive-log
    println(apiKey)
  }

  def safe(): Unit = {
    // ok: scala-sensitive-log
    logger.info("user logged in")
    // ok: scala-sensitive-log
    println("request processed")
    // ok: scala-sensitive-log
    logger.debug("redacted")
  }
}
